src/Xml/TypeNode.php

```diff
@@ -230,6 +220,28 @@ public function isComplex()
     }
 
     /**
+     * Returns whether the type is an array.
+     *
+     * @return bool If the type is an array.
+     */
+    public function isArray()
+    {
+        $parts = $this->getParts();
+
+        if (!$this->isComplex()
+            || count($parts) !== 1 || (
+                substr($this->name, -2, 2) !== '[]' &&
+                substr($this->name, 0, 7) !== 'ArrayOf')
+        ) {
+            return false;
+        }
+
+        $varType = reset($parts);
+
+        return substr($varType, -2, 2) == '[]';
```

Comment:
This change increases the complexity of the `isArray()` method quite a bit. I would appreciate an inline comment explaining why such a check is necessary.